performance/collector/es.py

```diff
@@ -0,0 +1,114 @@
+
+
+@exporter.configure(name="es")
+class ESExporter(exporter.Exporter):
+
+    def validate(self):
+        """Validate connection string.
+
+        The format of connection string in ElasticSearch plugin is
+            es:///<path>.<region>
+        """
+        LOG.debug("Region: %s " % self.region)
+        LOG.debug("URI: %s " % self.url)
+
+        parse_obj = six.moves.urllib.parse.urlparse(self.connection_string)
+
+        if self.connection_string is None or parse_obj.path == "":
+            raise exceptions.InvalidConnectionString(
+                "It should be `es:///<ip_or_hostname>.<region>`.")
+        if not self.region:
+            raise exceptions.InvalidConnectionString(
+                "Invalid or undefined region")
+
+    def __init__(self, connection_string):
+        super(ESExporter, self).__init__(connection_string)
+        self.path = os.path.expanduser(six.moves.urllib.parse.urlparse(
+            connection_string).path[1:])
+        self.region = connection_string.split(".")[-1]
+        self.url = (
+            connection_string.replace('.' + self.region, '').split('/')[-1])
+        self.proto = 'http://'
+        self.validate()
+
+    def export(self, uuid):
+        """Export results of the task to the file.
+
+        :param uuid: uuid of the task object
+        """
+
+        task = api.Task.get(uuid)
+
+        LOG.debug("Got the task object by it's uuid %s. " % uuid)
+
+        task_results = [{"key": x["key"], "result": x["data"]["raw"],
+                         "sla": x["data"]["sla"],
+                         "hooks": x["data"].get("hooks"),
+                         "load_duration": x["data"]["load_duration"],
+                         "full_duration": x["data"]["full_duration"]}
+                        for x in task.get_results()]
+
+        es_data_list = []
+        if task_results:
+            LOG.debug("Got the task %s results." % uuid)
+            for task_result in task_results:
+                for result in task_result['result']:
+                    if not result['error']:
+                        atomic_actions = result['atomic_actions']
+                        for k, v in six.iteritems(atomic_actions):
+                            es_data = {
+                                'region': self.region,
+                                'metric': k,
+                                'value': v,
+                                'timestamp': datetime.datetime.now().strftime(
+                                    "%Y-%m-%dT%H:%M:%S+00:00")
+                            }
+                            es_data_list.append({'index': {}})
+                            es_data_list.append(es_data)
+        else:
+            msg = ("Task %s results would be available when it will "
+                   "finish." % uuid)
+            raise exceptions.RallyException(msg)
+
+        LOG.debug("ES Data: \n %s ",
+                  '\n'.join([json.dumps(x) for x in es_data_list]))
```

Comment:
this is very heavy operation, please make it only once (instead of 104 and 109 lines)